### PR TITLE
Lab08

### PR DESCRIPTION
**Pull request by bodnarira**

### CODE REVIEW

File: Lab08.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import os
+
+class Dir:
+    def __init__ (self, path):
+        self.path = path
+    def duplicates (self):
```
Comment (polyakovartem) [inline Lab08.py]: please split this function, the logic is to complex for one function

File: Lab08.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                    if added == False:
+                        result.append([])
+                        result[len(result)-1].append(list_of_files[i])
+                        added = True
+                    result[len(result)-1].append(list_of_files[j])
+
+        return result
+
+dir = Dir("D:/Ira/Python/new/")
+print(dir.duplicates())
```
Comment (polyakovartem) [inline Lab08.py]: where is __main__ ?

**Review by polyakovartem — CHANGES_REQUESTED** [review]

**Review by polyakovartem — CHANGES_REQUESTED** [review]
by the way, I've already seen this code by your colleagues, could you please explain how did it happen?

**Comment by bodnarira** [comment]
> by the way, I've already seen this code by your colleagues, could you please explain how did it happen?

Send me a link of this code you have seen.

File: Lab08.py (hunk 2 above)
Comment (bodnarira) [reply Lab08.py]: ok

**Review by polyakovartem — CHANGES_REQUESTED** [review]
could you kindly explain why you did not fix other comments?

File: Lab08.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                    if added == False:
+                        result.append([])
+                        result[len(result)-1].append(list_of_files[i])
+                        added = True
+                    result[len(result)-1].append(list_of_files[j])
+
+        return result
+
+if __name__ == '__main__':
+    dir = Dir("D:/Ira/Python/new/")
```
Comment (polyakovartem) [inline Lab08.py]: it's not the best idea to use absolute paths in the code

File: Lab08.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import os
+
+class Dir:
+    def duplicates (path:str):
```
Comment (polyakovartem) [inline Lab08.py:4]: split this function into several smaller functions

**Review by polyakovartem — CHANGES_REQUESTED** [review]

**Pull request closed without merging**